src/CodeGen_Hexagon.cpp

```diff
@@ -2232,6 +2232,28 @@ void CodeGen_Hexagon::visit(const Call *op) {
       Value *Call = CallLLVMIntrinsic(F, Ops);
       value = convertValueType(Call, llvm_type_of(op->type));
       return;
+    } else if (op->name == Call::interleave_vectors) {
+      if (isDblVector(op->type, native_vector_bits()) &&
+                      op->args.size() == 2) {
+        Expr arg0 = op->args[0];
+        Expr arg1 = op->args[1];
+        if ((arg0.type() == arg1.type()) &&
+            op->type.element_of() == arg0.type().element_of()) {
+          internal_assert(op->type.lanes() == 2*arg0.type().lanes());
+          std::vector<Value *> Ops;
+          Ops.push_back(codegen(arg1));
+          Ops.push_back(codegen(arg0));
+          int scalar = -1 * op->type.bytes();
+          Value *scalar_value = codegen(IntImm::make(Int(32), scalar));
```

Comment:
You should be able to just say codegen(scalar), no need to explicitly construct an IntImm. It's a bit easier to read IMO.